example/fib/main.go

```diff
@@ -60,13 +60,15 @@ func main() {
 	// Write telemetry data to a file.
 	f, err := os.Create("traces.txt")
 	if err != nil {
-		l.Fatal(err)
+		l.Println(err)
+		return
```

Comment:
we do not set the exit code

We may consider doing something like the following in the begining of the `main` function

```golang
var exitCode int
defer os.Exit(exitCode)
```

and then here before `return` we could add `exitCode = 1`

Reply:
In a second though, I think it is safe here to do `l.Fata(err)` as it was before since there are no resources to free using a deferred function yet at this stage.

On the other hand, I think it is also safe to do `exp, _ := newExporter(f)` since the returned **_err_** will always be **_nil_** no matter what as explained below. 

`newExporter()` calls `stdouttrace.New()` which internally calls `newConfig()` and the latter is the only one in that chain of calls that could return a value to the second returned value (error), and further examination to `newConfig()` shows that it always returns **_nil_** value for **error**.

https://github.com/open-telemetry/opentelemetry-go/blob/4e763472eee5ef03b97274694d2d1b07f0b3cd11/exporters/stdout/stdouttrace/trace.go#L33

https://github.com/open-telemetry/opentelemetry-go/blob/4e763472eee5ef03b97274694d2d1b07f0b3cd11/exporters/stdout/stdouttrace/config.go#L52

Is it by design in order to have parity with other packages or should we create another PR?

Reply:
@pellared are your concerns addressed here?

Reply:
Not addressed. But it is not a blocker for me😉 

We could also consider using `l.Panic(err)` or `panic(err)` instead which takes care of deferred functions and set  exit code.  If we use `panic(err)` we could even get rid of `l := log.New(os.Stdout, "", 0)`.

Related PR: https://github.com/open-telemetry/opentelemetry-go/pull/4090